service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -756,6 +756,34 @@ fi
 chown ${LOGIN_USER}:${LOGIN_USER} "${USER_BASHRC}"
 chown ${LOGIN_USER}:${LOGIN_USER} "${USER_BASH_PROFILE}"
 
+#######################################
+# Restart proxy to pick up the new env
+######################################
+readonly APP_PROXY=$(get_metadata_value "instance/attributes/terra-app-proxy")
+readonly PROXY_ENV="/opt/deeplearning/proxy.env"
+if [[ -n "${APP_PROXY}" ]]; then
+    emit "Using custom Proxy Agent"
+    RESOURCE_ID=$(get_metadata_value "instance/attributes/terra-resource-id")
+    NEW_PROXY="https://${APP_PROXY}"
+    NEW_PROXY_URL="${RESOURCE_ID}.${APP_PROXY}"
+    # escape all the orrurences of forward slash
+    ESCAPED_NEW_PROXY=$(sed 's/[\/&]/\\&/g' <<< "${NEW_PROXY}")
+    sed -i "s/^PROXY_REGISTRATION_URL=.*/PROXY_REGISTRATION_URL=${ESCAPED_NEW_PROXY}/" "${PROXY_ENV}"
+    sed -i "s/^PROXY_URL=.*/PROXY_URL=${NEW_PROXY_URL}/" "${PROXY_ENV}"
+    sed -i "s/^BACKEND_ID=.*/BACKEND_ID=${RESOURCE_ID}/" "${PROXY_ENV}"
+    systemctl restart notebooks-proxy-agent.service
+    emit "Proxy Agent service restarted"
+    INSTANCE_NAME=$(get_metadata_value "instance/name")
+    INSTANCE_ZONE="/"$(get_metadata_value "instance/zone")
+    INSTANCE_ZONE="${INSTANCE_ZONE##/*/}"
+    gcloud compute instances add-metadata "${INSTANCE_NAME}" \
+                  --metadata proxy-url="${NEW_PROXY_URL}" --zone "${INSTANCE_ZONE}"
+    emit "Overwrote proxy-url metadata"
+    # escape all the occurence of '.'
```

Comment:
```
    ...
    emit "Overwrote proxy-url metadata"

    # Since the field we are writing is the NOTEBOOK_CONFIG is a regular expression pattern,
    # we need to be sure to escape regex characters, notably the periods.
    ESCAPED_NEW_PROXY_URL=$(echo "${NEW_PROXY_URL}" | sed -r "s/\./\\\./g")

    cat << EOF >> "${NOTEBOOK_CONFIG}"

    c.ServerApp.allow_origin_pat += "|(^https://${ESCAPED_NEW_PROXY_URL}$)"

EOF
```